WalletWasabi/Extensions/TaskExtensions.cs

```diff
@@ -78,12 +78,23 @@ public static async Task WithAwaitCancellationAsync(this Task me, TimeSpan timeo
 			=> await me.WithAwaitCancellationAsync(ThreadingHelpers.Cancelled, (int)timeout.TotalMilliseconds).ConfigureAwait(false);
 
 		public static async Task<TResult> RunAsScheduledAsync<TResult>(this Task<TResult> task, DateTimeOffset when, CancellationToken cancellationToken)
+		{
+			await WaitUntilAsync(when, cancellationToken).ConfigureAwait(false);
+			return await task.ConfigureAwait(false);
+		}
+
+		public static async Task RunAsScheduledAsync(this Task task, DateTimeOffset when, CancellationToken cancellationToken)
+		{
+			await WaitUntilAsync(when, cancellationToken).ConfigureAwait(false);
+			await task.ConfigureAwait(false);
+		}
+
+		private static async Task WaitUntilAsync(DateTimeOffset when, CancellationToken cancellationToken)
 		{
 			var timeToWait = when - DateTimeOffset.UtcNow;
 			var fixedTimeToWait = timeToWait < TimeSpan.Zero ? TimeSpan.Zero : timeToWait;
 
 			await Task.Delay(fixedTimeToWait, cancellationToken).ConfigureAwait(false);
-			return await task.ConfigureAwait(false);
```

Comment:
This feels slightly simpler:

```csharp
private static async Task WaitUntilAsync(DateTimeOffset when, CancellationToken cancellationToken)
{
	var timeToWait = when - DateTimeOffset.UtcNow;
    if (timeToWait > TimeSpan.Zero) 
    {
	     await Task.Delay(timeToWait, cancellationToken).ConfigureAwait(false);
    }
}
```

and we wouldn't wait when `fixedTimeToWait = TimeSpan.Zero`.

Reply:
Yes, it is. I will add this suggestion when I find the opportunity. I preferred to merge it asap but the suggestion is not ignored.